sphinx/ext/autodoc/__init__.py

```diff
@@ -696,6 +696,11 @@ def is_filtered_inherited_member(name: str, obj: Any) -> bool:
                     RemovedInSphinx80Warning, stacklevel=2,
                 )
 
+            # filter a member having invalid name
```

Comment:
Should we really filter out the 'invalid' names? because, they are still valid in Python although we need to access them using `getattr` and not via dots. 

Should we perhaps add some escape sequences and modify the parsers accordingly? (in which case the PR will likely be much bigger). If we go with your approach, I'll suggest adding some documentation saying that exotic member names will not be rendered (for now).

Reply:
I feel showing `Foo.invalid;attr` in API docs is not good because users can't access such an invalid attribute. As you said, users can access it in other ways. But it's better to mention the better approach in the document, I think.

So I prefer to add a note to the doc of autodoc. What do you think?

I noticed this would be a breaking change. So it would be better to merge this into the next major version. Please let me know your thoughts.

Reply:
Yes, let us say that in the docs, namely, we don't support rendering those weird attributes. 

> I noticed this would be a breaking change. So it would be better to merge this into the next major version. Please let me know your thoughts.

Currently, do we even manage to render those weird fields? If so, I think we don't beautifully render them so I don't know whether it's really a "breaking" change or not. Neverthless, people may perhaps have some hacky approaches so to be on the safe side, we could merge this PR in 8.0 and then maybe work for a better rendering.